app/src/main/java/net/aiscope/gdd_app/network/FirestoreUtil.kt

```diff
@@ -0,0 +1,15 @@
+package net.aiscope.gdd_app.network
+import com.google.firebase.firestore.FirebaseFirestore
+
+class FirestoreUtil {
```

Comment:
Hello, why do you create a new instance for Firebase Firestore? You could use FirebaseRemoteStorage to store the sample, that has an instance of the Firebase FireStore

Reply:
I could not find firebaseFirestore instance in FirebaseRemoteStorage. FirebaseModule exists, but I am not sure how to use it. Unless I move all DB operations to SampleRepositorySharedPreference ?

PS: To understand things better, I have simplified the code, moved all DB operations to one SampleViewModel and commented everything else. Where should this code be best moved to ?

Reply:
@simtop  I have added a new repository to store sample in firestore. Currently it does not implement  samplerepository interface.  feel it's too early for that.  Cleaned up a bit. Let me know.